roles/swapfile/tasks/main.yml

```diff
@@ -26,8 +26,13 @@
     fstype: swap
     state: present
 
-- name: swapon {{ swapfile_path }}
+- name: get active swap
+  command: swapon --summary | grep '^{{ swapfile_path }}'
+  register: swapfile_active
+
+- name: activate {{ swapfile_path }}
   command: swapon {{ swapfile_path }}
+  when: swapfile_active.rc != 0
```

Comment:
Looks like this doesn't work. Is there a native way to grep in ansible?

Reply:
You can do `swapfile_active | failed`

Reply:
Doesn't appear to help. Somehow the activation is still skipped so the command somehow succeeds. Perhaps it needs to be quoted because of the pipe?